### PR TITLE
Added OOP approach in accesing controller

### PR DESCRIPTION
**Pull request by emmanueljustin**

### CODE REVIEW

File: app/Http/Controllers/TodoController.php [hunk 1]
```diff
@@ -2,15 +2,23 @@
...
 class TodoController extends Controller
 {
+
+    protected $todoService;
+
+    public function __construct(TodoService $todoService) {
+        $this->todoService = $todoService;
+    }
+
     public function index()
```
Comment (hipe01mark) [inline app/Http/Controllers/TodoController.php]: add doc block
add return type

File: app/Http/Controllers/TodoController.php [hunk 2]
```diff
@@ -28,7 +36,7 @@ public function index()
 
     public function store(TodoRequest $payload)
```
Comment (hipe01mark) [inline app/Http/Controllers/TodoController.php]: add doc block
add return type

File: app/Http/Controllers/TodoController.php [hunk 3]
```diff
@@ -39,7 +47,7 @@ public function store(TodoRequest $payload)
 
     public function update(TodoRequest $payload, string $id)
```
Comment (hipe01mark) [inline app/Http/Controllers/TodoController.php]: add doc block
add return type

File: app/Http/Controllers/TodoController.php [hunk 4]
```diff
@@ -98,9 +104,16 @@ public function update(TodoRequest $payload, string $id)
 
     function deleteTodos(Request $payload) {
```
Comment (hipe01mark) [inline app/Http/Controllers/TodoController.php]: add doc block
add return type

File: app/Http/Requests/BaseRequest.php [hunk 5]
```diff
@@ -30,7 +30,10 @@ public function rules(): array
 
     public function failedValidation(Validator $validator)
```
Comment (hipe01mark) [inline app/Http/Requests/BaseRequest.php:31]: add doc block
add return type

File: app/Repositories/TodoRepository/BaseTodoRepository.php [hunk 6]
```diff
@@ -0,0 +1,14 @@
...
+use App\Models\Todo;
+use Illuminate\Support\Collection;
+
+interface BaseTodoRepository {
+    public function getAll() : Collection;
+    public function findById(string $id) : Todo;
+    public function create(array $payload) : Todo;
+    public function updateData(array $payload, string $id) : ?Todo;
+    public function batchDelete(array $ids) : bool;
+}
```
Comment (hipe01mark) [inline app/Repositories/TodoRepository/BaseTodoRepository.php]: Add new line every end of line of the file

File: app/Repositories/TodoRepository/TodoRepository.php [hunk 7]
```diff
@@ -0,0 +1,40 @@
...
+        return null;
+    }
+
+    public function batchDelete(array $ids) : bool {
+        if (!empty($ids)) {
+            return Todo::whereIn("id", $ids)->delete();
+        }
+        return false;
+    }
+}
```
Comment (hipe01mark) [inline app/Repositories/TodoRepository/TodoRepository.php]: Add new line every end of line of the file
Add docbloc every method

File: app/Services/TodoService.php [hunk 8]
```diff
@@ -0,0 +1,36 @@
...
+    }
+
+    public function updateData(array $payload, string $id) : ?Todo {
+        return $this->repo->updateData($payload, $id);
+    }
+
+    public function batchDelete(array $ids) : bool {
+        return $this->repo->batchDelete($ids);
+    }
+}
```
Comment (hipe01mark) [inline app/Services/TodoService.php]: Add new line every end of line of the file
Add docbloc every method

**Review by karljerico — APPROVED** [review]
LGTM 🙇

**Review by mikco-hipe — APPROVED** [review]
LGTM

File: apache.conf [hunk 9]
```diff
@@ -0,0 +1,18 @@
...
+    </Directory>
+
+    ErrorLog ${APACHE_LOG_DIR}/error.log
+    CustomLog ${APACHE_LOG_DIR}/access.log combined
+
+    # Optional: CORS headers for APIs
+    Header set Access-Control-Allow-Origin "*"
+    Header set Access-Control-Allow-Methods "GET, POST, OPTIONS"
+    Header set Access-Control-Allow-Headers "Authorization, Content-Type"
+</VirtualHost>
```
Comment (hipe01mark) [inline apache.conf:18]: New line at the end of the file

**Review by hipe01mark — APPROVED** [review]
LGTM 🙇

File: apache.conf (hunk 9 above)
Comment (emmanueljustin) [reply apache.conf:18]: copy sir

**Pull request merged**